platform/features/table/src/controllers/TableController.js

```diff
@@ -0,0 +1,214 @@
+            this.changeListeners = [];
+
+            $scope.rows = [];
+
+            // Subscribe to telemetry when a domain object becomes available
+            this.$scope.$watch('domainObject', function(domainObject){
+                if (!domainObject)
+                    return;
+
+                self.subscribe();
+                self.registerChangeListeners();
+            });
+
+            // Unsubscribe when the plot is destroyed
+            this.$scope.$on("$destroy", this.destroy.bind(this));
+        }
+
+        TableController.prototype.registerChangeListeners = function() {
+            //Defer registration of change listeners until domain object is
+            // available in order to avoid race conditions
+
+            this.changeListeners.forEach(function (listener) {
+                return listener && listener();
+            });
+            this.changeListeners = [];
+            // When composition changes, re-subscribe to the various
+            // telemetry subscriptions
+            this.changeListeners.push(this.$scope.$watchCollection('domainObject.getModel().composition', this.subscribe.bind(this)));
+
+            //Change of bounds in time conductor
+            this.changeListeners.push(this.$scope.$on('telemetry:display:bounds', this.subscribe.bind(this)));
+
+        };
+
+        /**
+         * Release the current subscription (called when scope is destroyed)
+         */
+        TableController.prototype.destroy = function () {
+            if (this.handle) {
+                this.handle.unsubscribe();
+                this.handle = undefined;
+            }
+        };
+
+        /**
+         Create a new subscription. This is called when
+         */
+        TableController.prototype.subscribe = function() {
+            var self = this;
+
+            /*if (this.pending){
+                return;
+            }*/
+            //this.pending = true;
+
+            if (this.handle) {
+                this.handle.unsubscribe();
+            }
+
+            this.$scope.rows = [];
+
+            //Noop because not supporting realtime data right now
+            function noop(){
+                //self.pending = false;
+            }
+
+            this.handle = this.$scope.domainObject && this.telemetryHandler.handle(
+                    this.$scope.domainObject,
+                    noop,
+                    true // Lossless
+                );
+
+            this.handle.request({}, this.addHistoricalData.bind(this));
+
+            this.setup();
+        };
+
+        /**
+         * Add any historical data available
+         */
+        TableController.prototype.addHistoricalData = function(domainObject, series) {
+            var i;
+            //this.pending = false;
+            for (i=0; i < series.getPointCount(); i++) {
+                this.updateRows(domainObject, this.handle.makeDatum(domainObject, series, i));
```

Comment:
Okay, so some interesting thoughts on telemetry handle, datums, and limits.  First off, I'm not a big fan of telemetry handles/handlers/subscriptions-- would prefer we talk directly to the telemetry API (general idea: less interfaces).   That's my biggest comment here: what does it take to remove telemetry handlers from this code?

On the VISTA side, the TelemetrySeries implements a `getDatum` method which can be used in place of `makeDatum` with minor differences-- it does not parse the underlying values.  In VISTA, the datum values used for limits evaluation are not ranges or domains, so they won't come through if you use `makeDatum`, and alarm validation won't work.  This is a difference we need to figure out how to resolve.  I'd push for adding a `getDatum` method to all telemetry series to use instead of `makeDatum`.

`updateRows` could be changed to take a domainObject and a telemetrySeries.  From there, it should be able to pull out only the range/domain values it needs (based on column configuration)-- and not the entire datum.  Additionally, it seems wasteful to re-evaluate limits for each cell in a row, when the logic is essentially the same.  Can we assign the CSS class to the entire row and not the individual cells?